pypowsybl/perunitnetwork.py

```diff
@@ -0,0 +1,615 @@
+        three_windings_transformers['x2'] /= three_windings_transformers['nominal_v'] ** 2 / self.sn
+        three_windings_transformers['g2'] /= self.sn / three_windings_transformers[
+            'nominal_v'] ** 2
+        three_windings_transformers['b2'] /= self.sn / three_windings_transformers[
+            'nominal_v'] ** 2
+        three_windings_transformers['r3'] /= three_windings_transformers['nominal_v'] ** 2 / self.sn
+        three_windings_transformers['x3'] /= three_windings_transformers['nominal_v'] ** 2 / self.sn
+        three_windings_transformers['g3'] /= self.sn / three_windings_transformers[
+            'nominal_v'] ** 2
+        three_windings_transformers['b3'] /= self.sn / three_windings_transformers[
+            'nominal_v'] ** 2
+        three_windings_transformers['rated_u1'] /= three_windings_transformers['nominal_v']
+        three_windings_transformers['rated_u2'] /= three_windings_transformers['nominal_v']
+        three_windings_transformers['rated_u3'] /= three_windings_transformers['nominal_v']
+        return three_windings_transformers.drop(['nominal_v'],
+                                                axis=1)
+
+    def get_shunt_compensators(self) -> _pd.DataFrame:
+        """ Get shunt compensators as a ``Pandas`` data frame.
+
+        Returns:
+            a shunt compensators data frame
+        """
+        shunt_compensators = self.get_elements(_pypowsybl.ElementType.SHUNT_COMPENSATOR)
+        shunt_compensators = _pd.merge(shunt_compensators, self.get_voltage_levels()['nominal_v'],
+                                       left_on='voltage_level_id', right_index=True)
+        shunt_compensators['p'] /= self.sn
+        shunt_compensators['q'] /= self.sn
+        shunt_compensators['g'] /= self.sn / shunt_compensators['nominal_v'] ** 2
+        shunt_compensators['b'] /= self.sn / shunt_compensators['nominal_v'] ** 2
+        return shunt_compensators.drop('nominal_v', axis=1)
+
+    def get_dangling_lines(self) -> _pd.DataFrame:
+        """ Get dangling lines as a ``Pandas`` data frame.
+
+        Returns:
+            a dangling lines data frame
+        """
+        dangling_lines = self.get_elements(_pypowsybl.ElementType.DANGLING_LINE)
+        dangling_lines['p'] /= self.sn
+        dangling_lines['q'] /= self.sn
+        dangling_lines['p0'] /= self.sn
+        dangling_lines['q0'] /= self.sn
+        dangling_lines = _pd.merge(dangling_lines, self.get_voltage_levels()['nominal_v'],
+                                   left_on='voltage_level_id', right_index=True)
+        dangling_lines['i'] /= self.sn * 10 ** 3 / (
+                self.sqrt3 * dangling_lines['nominal_v'])
+        dangling_lines['r'] /= dangling_lines['nominal_v'] ** 2 / self.sn
+        dangling_lines['x'] /= dangling_lines['nominal_v'] ** 2 / self.sn
+        dangling_lines['g'] /= dangling_lines['nominal_v'] ** 2 / self.sn
+        dangling_lines['b'] /= dangling_lines['nominal_v'] ** 2 / self.sn
```

Comment:
It seems incorrect for g and b. Per unitage for admittance cannot be the same as for impedance. It should be dangling_lines['g'] /= self.sn / dangling_lines['nominal_v'] ** 2. Same for b.